### ISSUE TITLE
[Improvement] Supplier Scorecard - Calculate only with active transactions on period

### ISSUE THREAD

**Issue by xickomesquita** [frappe/erpnext#31597]
There isn't a way to evaluate suppliers only when it have active transactions.

when we use this calculation:

({total_accepted_items} / {total_received_items})*100
if {total_received_items} > 0
else 100

the 'else' its a poor workaround.

The supplier X had some transactions with my company only in months 1,2,4 and 7. The system is actually generating these scores:

![image](https://user-images.githubusercontent.com/38265198/179004973-b014c0e7-f581-490d-a8a3-a36f7a19af0c.png)

Instead of test only the months that really exist some purchase:

![image](https://user-images.githubusercontent.com/38265198/179005227-ed16b57b-ebdd-47cd-99dc-8e983a4215a7.png)

I believe that is necessary to change the .py file, to include a criteria to don't create a new scorecard, if it doen't have active transactions to the supplier.

**Comment by bosue** [comment]
You‘re completely right. These scorecards could be such a powerful tool, but need quite some work, because the details need be done right. I‘ll be working on this the next months.
